sympy/polys/matrices/tests/test_sdm.py

```diff
@@ -274,3 +274,12 @@ def test_SDM_particular():
     A = SDM({0:{1:QQ(1)}}, (2, 2), QQ)
     Apart = SDM({0:{1:QQ(1)}}, (1, 2), QQ)
     assert A.particular() == Apart
+
+def test_SDM_applyfunc():
+    A = SDM.eye(4, ZZ)
+    assert A.applyfunc(lambda x: ZZ(2) * x) == ZZ(2) * SDM.eye(4, ZZ)
+    assert A.applyfunc(lambda x: ZZ(0)) == SDM.zeros((4, 4), ZZ)
+
+def test_SDM_applyfunc_nonzero():
+    A = SDM.eye(3, ZZ)
+    assert A.applyfunc_nonzero(lambda x: x + QQ(1, 2), QQ) == QQ(3, 2) * SDM.eye(3, QQ)
```

Comment:
These tests should also include examples where nonzero elements become zero and vice versa